devito/data/decomposition.py

```diff
@@ -241,6 +249,24 @@ def convert_index(self, *args, rel=True):
                 if glb_idx_max is not None and glb_idx_max < 0:
                     glb_idx_max = self.glb_max + glb_idx_max + 1
                 # -> Do the actual conversion
+                # First check for the special case of a slice with negative step
+                if isinstance(glb_idx, slice) and glb_idx.step < 0:
+                    if glb_idx_max is None or glb_idx_max > self.loc_abs_max:
+                        loc_max = self.loc_abs_max - base
+                    elif glb_idx_max < self.loc_abs_min:
+                        return retfunc(-1, -3)
+                    else:
+                        loc_max = glb_idx_max - base
+                    if glb_idx_min is not None:
```

Comment:
likewise, these look very similar to the block below, except for the -1 ...

Reply:
This `if glb_idx_min is not None:` is also different which makes compacting things a bit tricky?

Reply:
yep agreed